src/NuGet.Clients/PackageManagement.VisualStudio/EnvDTEProjectAdapter.cs

```diff
@@ -164,42 +156,21 @@ public bool IsLegacyCSProjPackageReferenceProject
 
                 if (!_isLegacyCSProjPackageReferenceProject.HasValue)
                 {
-                    var restoreProjectStyle = string.Empty;
-
-                    var hasRestoreProjectStyle = _context
-                        .MSBuildProperties
-                        .TryGetValue(ProjectSystemProviderContext.RestoreProjectStyle, out restoreProjectStyle)
-                        && !string.IsNullOrEmpty(restoreProjectStyle);
-
-                    if (!hasRestoreProjectStyle)
+                    // A legacy CSProj can't be CPS, must cast to VSProject4 and *must* have at least one package
+                    // reference already in the CSProj. In the future this logic may change. For now a user must
+                    // hand code their first package reference. Laid out in longhand for readability.
+                    if (AsIVsHierarchy?.IsCapabilityMatch("CPS") ?? true)
                     {
-                        // A legacy CSProj can't be CPS, must cast to VSProject4 and *must* have at least one package
-                        // reference already in the CSProj. In the future this logic may change. For now a user must
-                        // hand code their first package reference. Laid out in longhand for readability.
-                        if (AsIVsHierarchy?.IsCapabilityMatch("CPS") ?? true)
-                        {
-                            _isLegacyCSProjPackageReferenceProject = false;
-                        }
-                        else if (AsVSProject4 == null ||
-                                (AsVSProject4.PackageReferences?.InstalledPackages?.Length ?? 0) == 0)
-                        {
-                            _isLegacyCSProjPackageReferenceProject = false;
-                        }
-                        else
-                        {
-                            _isLegacyCSProjPackageReferenceProject = true;
-                        }
+                        _isLegacyCSProjPackageReferenceProject = false;
                     }
-                    else if (restoreProjectStyle.Equals(ProjectStyle.PackageReference.ToString(), 
-                        StringComparison.OrdinalIgnoreCase))
+                    else if (AsVSProject4 == null ||
+                        (AsVSProject4.PackageReferences?.InstalledPackages?.Length ?? 0) == 0)
```

Comment:
> we don't know whether they will really need it or not since going forward

It will. To the user there is no difference for this scenario between NETCore and Legacy. Shipping a new feature in a broken state like this because no one is using it yet doesn't seem like a valid reason :smile:

Also, making PackageReference the default has no impact here. There are three states:
1. packages.config
1. PackageReference
1. non-NuGet project

Ignoring RestoreStyleProject means that all projects would incorrect become either 2 or 3 instead of the current behavior of 1 or 3. 

With packages.config there are no transitive dependencies so this isn't an issue. RestoreStyleProject=PackageReference is a switch to turn on transitive behavior without having to add PackageReferences, a very common scenario.